testcontainers/testcontainers-dotnet
# feat: Use built-in PEM certificate import on .NET 6 and onwards

## What does this PR do?

This pull requests uses the built-in PEM certificate import on .NET 6 onwards and updates the package references conditionally on the target framework.

`BouncyCastle.Cryptography`, `Microsoft.Bcl.AsyncInterfaces` and `System.Text.Json` are now referenced only when targeting .NET Standard (2.0 and 2.1) since they are not needed anymore when targeting .NET 6 onwards.

## Why is it important?

This allows to drop the BouncyCastle.Cryptography dependency (almost 7 MB) when targeting net6.0 and net8.0.

## Related issues

None, I just saw that .NET 6 and .NET 8 support was added recently and took the opportunity to reduce the dependencies.

## How to test this PR

Just run the existing tests and verify that nothing is broken. Note that some [care had to be taken on Windows](https://stackoverflow.com/questions/72096812/loading-x509certificate2-from-pem-file-results-in-no-credentials-are-available/72101855#72101855).
